### DISCUSSION TITLE
Is it possible to include more parameters in the metadata of generated images

### DISCUSSION THREAD

**Discussion by toyssamurai** [vladmandic/sdnext#1997] [Ideas]
With the continuous addition of new features to Stable Diffusions/SD.Next, the parameters included in the metadata of the generated images have become inadequate for accurately reproducing an image. This inadequacy is particularly pronounced when it comes to Img2Img/Inpaint functionality. Essential parameters like denoising strength, denoise start, and guidance rescale have all been omitted. Is it possible to include more parameters in the metadata of generated images to address this limitation?

**Comment by vladmandic** [comment]
yes, its possible. 
it would be even better if you enumerate all the settings needed exactly.

**Reply by toyssamurai** [reply]
Sure I can do that, but instead of maintaining a list manually, wouldn't it be a better way to implement the feature if the codes dynamically collect all the values from the entire workflow?

**Reply by vladmandic** [reply]
Not really. Some items just can't be included - like img2img init image is pixel data, can't place entire pixel data of input image in metadata of output image.

**Reply by toyssamurai** [reply]
Here are what I think are missing currently:

Txt2Img:
Guidance rescale (not sure if it's not included simply because the parameter isn't used by the refiner)

Img2Img/Inpaint:
Resize mode
Denoising strength
Denoise start
Image CFG Scale (not sure if it's used at all)
Guidance Rescale (not sure if it's used at all)

I am not sure if the following parameters are needed because they are closely related to the inpaint mask, but since the mask blur and the original image's hash are already included, I guess they should be included, too?

Mask mode
Inpaint area
Masked content
Mask Padding

**Comment by vladmandic** [comment]
still work-in-progress but first major update is complete, see <https://github.com/vladmandic/automatic/wiki/Metadata> for details
